core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -874,20 +843,70 @@ object JsonAST {
     }
   }
 
+  sealed trait AppendContainer[T] {
+    def append(thing: Char): AppendContainer[T]
+    def append(thing: String): AppendContainer[T]
+    def toString: String
+  }
+  case class StringBuilderAppendContainer(builder: StringBuilder) extends AppendContainer[StringBuilder] {
+    override def append(thing: Char) = StringBuilderAppendContainer(builder.append(thing))
+    override def append(thing: String) = StringBuilderAppendContainer(builder.append(thing))
+    override def toString = builder.toString
+  }
+  case class AppendableAppendContainer(appendable: Appendable) extends AppendContainer[Appendable] {
+    override def append(thing: Char) = AppendableAppendContainer(appendable.append(thing))
+    override def append(thing: String) = AppendableAppendContainer(appendable.append(thing))
+    override def toString = appendable.toString
+  }
+
+  object RenderSettings {
+    val pretty = RenderSettings(2)
+    val compact = RenderSettings(0)
+  }
```

Comment:
Personal preference: I'd put the singleton after the definition of `RenderSettings`.